core/rawdb/accessors_chain.go

```diff
@@ -690,8 +699,12 @@ type receiptLogs struct {
 
 // DecodeRLP implements rlp.Decoder.
 func (r *receiptLogs) DecodeRLP(s *rlp.Stream) error {
+	blob, err := s.Raw()
+	if err != nil {
+		return err
+	}
 	var stored storedReceiptRLP
-	if err := s.Decode(&stored); err != nil {
+	if err := rlp.DecodeBytes(blob, &stored); err != nil {
```

Comment:
blob here is only useful if there are multiple decoding attempts of the same data. But there's only one, so the stream can be used right?

Reply:
oh good spot - it was required for the first approach I took to solve legacy tx, but can be simplified again now.